api/noc/errorCodeConverter.ts

```diff
@@ -0,0 +1,54 @@
+export function generateErrorMessageWithCode(error) {
+  let data
+  if (error.data && !error.data.code && error.data.message) {
+    const code = generateErrorCodeFromMessage(error.data.message)
+    data = {...error.data, code}
+  } else {
+    data = error.data
+  }
+  return {...error, data}
+}
+
+function generateErrorCodeFromMessage(message: string) {
+  if (message.toLowerCase().includes('Case ID can not be empty'.toLowerCase())) {
```

Comment:
1. This is 100% a case for a switch statement.
2. `message.toLowerCase()` is getting called for every check, which lends itself to being done before it gets tested - e.g., `const lowerMessage: string = message.toLowerCase();`
3. Why are you using hard-coded values rather than constants or an enum?
4. Why are they all _also_ being converted to lower case rather than being lower case to start with?

Reply:
change to switch now